hoomd/pytest/test_snapshot.py

```diff
@@ -212,3 +265,31 @@ def test_constraints(s):
         assert s.constraints.value.dtype == numpy.float64
         assert s.constraints.group.shape == (3, 2)
         assert s.constraints.group.dtype == numpy.uint32
+
+
+@skip_gsd
+def test_from_gsd_snapshot_empty(s, device):
+    gsd_snap = make_gsd_snapshot(s)
+    hoomd_snap = Snapshot.from_gsd_snapshot(gsd_snap, device.communicator)
+    assert_equivalent_snapshots(gsd_snap, hoomd_snap)
+
+
+@skip_gsd
+def test_from_gsd_snapshot_populated(s, device):
+    s.configuration.box = [10, 12, 7, 0.1, 0.4, 0.2]
+    s.particles.N = 5
+    s.particles.types = ['A', 'B']
+    s.bonds.N = 5
+    s.bonds.types = ['A', 'B']
+    s.angles.N = 5
+    s.angles.types = ['A', 'B']
+    s.dihedrals.N = 5
+    s.dihedrals.types = ['A', 'B']
+    s.impropers.N = 5
+    s.impropers.types = ['A', 'B']
+    s.pairs.N = 5
+    s.pairs.types = ['A', 'B']
+    s.constraints.N = 3
```

Comment:
Can we test that some of the array quantities get set correctly too?

Reply:
I like the changes you made :) I updated that test to follow a similar format and set all array quantities as well. Should I be more careful to keep the datatype the same or does it not matter for this test? (e.g. `s.particles.body` gets set to a random float which doesn't make sense...)

Reply:
mmm... I just went ahead and fixed it just in case